pkg/protocol/http1/client.go

```diff
@@ -167,6 +164,15 @@ func (c *HostClient) WantConnectionCount() (count int) {
 	return c.connsWait.len()
 }
 
+func (c *HostClient) ConnPoolState() config.ConnPoolState {
+	c.connsLock.Lock()
```

Comment:
Is it possible to use sync.RWMutex instead of sync.Mutex?

Reply:
There is seldom read scenario and no concurrent read scenario, so there is no need to change it.